plugins/services/src/js/reducers/serviceForm/Containers.js

```diff
@@ -222,18 +229,36 @@ function containersParser(state) {
               'labels'
             ], item.labels));
           }
-          memo.push(new Transaction(
-            ['containers', index, 'endpoints', endpointIndex, 'protocol'],
-            endpoint.protocol.join()
-          ));
+          endpoint.protocol.forEach(function (protocol) {
```

Comment:
![undefined-protocol](https://cloud.githubusercontent.com/assets/647035/22421952/2f4e2a9e-e6ea-11e6-9219-5dcb3669b955.gif)

Please add a default or `undefined` check for `protocol` as this otherwise results in errors as illustrated in the gif.